### ISSUE TITLE
Success Criterion 1.4.13: Content on Hover or Focus (Level AA)

### ISSUE THREAD

**Issue by nitedog** [w3c/wcag2ict#108]
From [Success Criterion 1.4.13](https://www.w3.org/TR/WCAG22/#content-on-hover-or-focus):

> Where receiving and then removing pointer hover or keyboard focus triggers additional content to become visible and then hidden, the following are true:
> 
> **Dismissible**
>    A [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an [input error](https://www.w3.org/TR/WCAG22/#dfn-input-error) or does not obscure or replace other content;
> 
> **Hoverable**
>    If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;
> 
> **Persistent**
>    The additional content remains visible until the hover or focus trigger is removed, the user dismisses it, or its information is no longer valid.
> 
> Exception: The visual presentation of the additional content is controlled by the user agent and is not modified by the author.
> 
> Note: Examples of additional content controlled by the user agent include browser tooltips created through use of the HTML [title attribute](https://www.w3.org/TR/html/dom.html#the-title-attribute).
> 
> Note: Custom tooltips, sub-menus, and other nonmodal popups that display on hover and focus are examples of additional content covered by this criterion.

##### Additional Guidance When Applying Success Criterion 1.4.13 to Non-Web Documents and Software:

- This applies directly as written, and as described in [Intent from Understanding Success Criterion 1.4.13](https://www.w3.org/WAI/WCAG22/Understanding/content-on-hover-or-focus.html) (also provided below), replacing: 
- Reference to "user agent" in the exception, similar to the [approach currently being discussed for SC 1.4.11](https://www.w3.org/2023/02/16-wcag2ict-minutes.html#item03)
- Reference to "browser" and "HTML" in the first note

With these substitutions, it would read:

> Where receiving and then removing pointer hover or keyboard focus triggers additional content to become visible and then hidden, the following are true:
> 
> **Dismissible**
>    A [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an [input error](https://www.w3.org/TR/WCAG22/#dfn-input-error) or does not obscure or replace other content;
> 
> **Hoverable**
>    If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;
> 
> **Persistent**
>    The additional content remains visible until the hover or focus trigger is removed, the user dismisses it, or its information is no longer valid.
> 
> Exception: The visual presentation of the additional content is controlled by the _operating system_ and is not modified by the author.
> 
> Note: Examples of additional content controlled by the _operating system_ include tooltips created through use of _component property names_.
> 
> Note: Custom tooltips, sub-menus, and other nonmodal popups that display on hover and focus are examples of additional content covered by this criterion.

**Comment by maryjom** [comment]
@nitedog Let me know when this is ready for survey.

**Comment by nitedog** [comment]
Yes, I think so.

**Comment by mitchellevan** [comment]
> controlled by the _operating system_

Non-web documents rely on a user agent, so we should say "controlled by the _user agent or operating system_". Or however we're saying this elsewhere — I think it's "controlled by the _user agent or platform software_" ...?

> created through use of _component property names_

Nit: I would say "created through use of _component properties_" — name-value pairs I presume

**Comment by maryjom** [comment]
Created [Survey for 1.4.13](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-Content-on-hover-or-focus/) Due 2 March. If you aren't a Task Force participant, you can view the [survey results](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-Content-on-hover-or-focus/results/).

**Comment by mraccess77** [comment]
It seems to me that use of "operating system" is too limited as this would be applied to non-web documents as well.   Exceptions might be controlled through a PDF viewer or Word Processor which are user agents.

**Comment by pday1** [comment]
I agree with the previous comments about "operating system" being a little restrictive

Would "operating system or platform software" could work if we wish to avoid using "user agent"?

**Comment by pday1** [comment]
Having just caught up with the latest updates to key terms, I think we should just use platform software, as the definition of this includes the operating system. 

See definition of "accessibility services of platform software" in key terms for more detail. Temp link is [https://deploy-preview-111--wcag2ict.netlify.app/#key-terms](preview of key terms) or [https://github.com/w3c/wcag2ict/pull/111/files](pull request #111 )

**Comment by mraccess77** [comment]
I agree with platform software - Section 508 it defines it as:
Platform Software. Software that interacts with hardware or provides services for other software. Platform software may run or host other software, and may isolate them from underlying software or hardware layers. A single software component may have both platform and non-platform aspects. Examples of platforms are: desktop operating systems; embedded operating systems, including mobile systems; Web browsers; plug- ins to Web browsers that render a particular media or format; and sets of components that allow other applications to execute, such as applications which support macros or scripting.

**Comment by loicmn** [comment]
There are several issues:

- I agree that we should use "platform software" instead of "operating system". It was the agreement reached in both Section 508 and EN 301 549. 
- I also agree that we need to keep "user agent" for non-web documents. 
- In the text explaining how this would apply to non-web ICT, the term substitutions should be explicit and detailed. They should allow anyone to "build" the adapted SC without looking at the part where we say "... it would read".

Taking this into account and the previous comments, here is my proposal.

For the  **additional guidance**:

> This applies directly as written, and as described in Intent from Understanding Success Criterion 1.4.13 (also provided below), replacing:
> 
> - "user agent" with "user agent (for non-web documents) or platform software (for non-web sofware)" in the exception and note 1
> - "browser tooltips" with "tooltips"
> - "the HTML title attribute" with "user interface object attributes"

That would imply that **the exception now reads**:

> Exception: The visual presentation of the additional content is controlled by the user agent (for non-web documents) or platform software (for non-web sofware) and is not modified by the author.

And the **note 1 would read:**

>Note: Examples of additional content controlled by the user agent (for non-web documents) or platform software (for non-web sofware) include tooltips created through use of user interface object properties.`

**Comment by maryjom** [comment]
Here is a version with proposals so far (as of 11 AM eastern on 1 March), incorporated and my additional edits to enumerate the word replacements consistent with how all of the other SC are handled.

From [Success Criterion 1.4.13](https://www.w3.org/TR/WCAG22/#content-on-hover-or-focus):

> Where receiving and then removing pointer hover or keyboard focus triggers additional content to become visible and then hidden, the following are true:
> 
> **Dismissible**
>    A [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an [input error](https://www.w3.org/TR/WCAG22/#dfn-input-error) or does not obscure or replace other content;
> 
> **Hoverable**
>    If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;
> 
> **Persistent**
>    The additional content remains visible until the hover or focus trigger is removed, the user dismisses it, or its information is no longer valid.
> 
> Exception: The visual presentation of the additional content is controlled by the user agent and is not modified by the author.
> 
> NOTE 
> Examples of additional content controlled by the user agent include browser tooltips created through use of the HTML [title attribute](https://www.w3.org/TR/html/dom.html#the-title-attribute).
> 
> NOTE
> Custom tooltips, sub-menus, and other nonmodal popups that display on hover and focus are examples of additional content covered by this criterion.

##### Guidance When Applying Success Criterion 1.4.13 to Non-Web Documents and Software:

This applies directly as written, and as described in [Intent from Understanding Success Criterion 1.4.13](https://www.w3.org/WAI/WCAG22/Understanding/content-on-hover-or-focus.html) (also provided below), replacing "user agent" with "user agent or platform software",  "browser tooltips" with "tooltips", and "the HTML title attribute" with "user interface object attributes".

With these substitutions, it would read:

> Where receiving and then removing pointer hover or keyboard focus triggers additional content to become visible and then hidden, the following are true:
> 
> **Dismissible**
>    A [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to dismiss the additional content without moving pointer hover or keyboard focus, unless the additional content communicates an [input error](https://w3c.github.io/wcag2ict/#dfn-input-error) or does not obscure or replace other content;
> 
> **Hoverable**
>    If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing;
> 
> **Persistent**
>    The additional content remains visible until the hover or focus trigger is removed, the user dismisses it, or its information is no longer valid.
> 
> Exception: The visual presentation of the additional content is controlled by the **[user agent or platform software]** and is not modified by the author.
> 
> Note: Examples of additional content controlled by the *[user agent or platform software]** include tooltips created through use of *[user interface object attributes]**.
> 
> Note: Custom tooltips, sub-menus, and other nonmodal popups that display on hover and focus are examples of additional content covered by this criterion.

**Comment by maryjom** [comment]
@ChrisLoiselle @pday1 Could one of you please implement the content from my comment above in GitHub? Whomever takes it first, unassign the other person so you aren't both working on it at the same time.

**Comment by pday1** [comment]
I've made a start on incorporating these changes in pull request PR #119

**Comment by maryjom** [comment]
@mraccess77 We don't have these terms defined in the document. "User interface object" is typical object oriented programming terminology for an instance/implementation of a UI component. "Attributes" are related to "properties" and in different programming languages, the "setting", "attribute" or "property" is the term used. In some programming languages there are differences between these.   I have to say trying to nail down very specific terms and definitions for these that fits all software programming languages is difficult.

**Comment by maryjom** [comment]
@mraccess77 This is simply a note on the exception that provides an example of what constitutes a tooltip whose display is controlled by a user agent or software platform. I don't see this as an expansion of the exception at all.

The original notes in WCAG were:

> Note: Examples of additional content controlled by the user agent include browser tooltips created through use of the HTML [title attribute](https://www.w3.org/TR/html/dom.html#the-title-attribute).
> 
> Note: Custom tooltips, sub-menus, and other nonmodal popups that display on hover and focus are examples of additional content covered by this criterion.

In WCAG, the first note is one example for HTML. We simply provided a non-web example similar to the original one, but in a non-web software context. Are you thinking we need to expand the example to include another example for how a tooltip is typically accomplished in a non-web document? If so, I'd prefer to have a new issue to track and resolve it since the AG WG approved the WCAG2ICT guidance as-is without changes in today's meeting.

**Comment by mraccess77** [comment]
@maryjom Ok, that seems like a reasonable approach.

**Comment by maryjom** [comment]
The AG WG approved the content on 11 April, so closing.
